pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1843,3 +1867,39 @@ func getProvisions(c client.Client) []*hivev1.ClusterProvision {
 	}
 	return provisions
 }
+
+func createSyncSetInstanceObj(syncCondType hivev1.SyncConditionType) *hivev1.SyncSetInstance {
+	ssi := &hivev1.SyncSetInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSyncsetInstanceName,
+			Namespace: constants.HiveNamespace,
+		},
+	}
+	ssi.Spec.ClusterDeployment.Name = testClusterName
+	ssi.Status = createSyncSetInstanceStatus(syncCondType)
+	return ssi
+}
+
+func createSyncSetInstanceStatus(syncCondType hivev1.SyncConditionType) hivev1.SyncSetInstanceStatus {
+	conditionTime := metav1.NewTime(time.Now())
+	var ssiStatus corev1.ConditionStatus
+	var condType hivev1.SyncConditionType
+	if syncCondType == hivev1.ApplyFailureSyncCondition {
+		ssiStatus = corev1.ConditionFalse
+		condType = syncCondType
+	} else {
+		ssiStatus = corev1.ConditionTrue
+		condType = hivev1.ApplySuccessSyncCondition
```

Comment:
Why is this set to `hivev1.ApplySuccsessSyncCondition` instead of `syncCondType`?

Reply:
Just keep the code more readable. Do you think it makes any difference?

Reply:
How is it more readable if the code does something unexpected like use a hardcoded condition type instead of the condition type passed in as an argument? As a reader, it makes me have to pause to reason about why the `ApplySuccessSyncCondition` condition type is special.